File: algorithms/2099.find-subsequence-of-length-k-with-the-largest-sum/solution.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Integer[]}
def max_subsequence(nums, k)
  res = []

  target = nums.sort[-k..]
  for i in 0...nums.size
    idx = target.index(nums[i])
    if idx
      res << nums[i]
      target.delete_at(idx)

      break if target.size == 0
    end
  end

  res
end

File: algorithms/2099.find-subsequence-of-length-k-with-the-largest-sum/solution.go
func maxSubsequence(nums []int, k int) []int {
	res := []int{}

	target := make([]int, len(nums))
	copy(target, nums)
	sort.Ints(target)
	target = target[len(nums)-k:]

	for _, n := range nums {
		idx := indexOf(n, target)
		if idx >= 0 {
			res = append(res, n)
			target = append(target[:idx], target[idx+1:]...)
		}
	}

	return res
}

func indexOf(n int, slice []int) int {
	for i, v := range slice {
		if n == v {
			return i
		}
	}

	return -1
}
